workflows/virtual-screening-nab/definition/main.nf

```diff
@@ -0,0 +1,154 @@
+nextflow.enable.dsl = 2
+
+// static data files are in nextflow.config
+workflow {
+    //Convert to files
+    if (params.smi_input_path[-1] == "/") {
+        smi_input_path = params.smi_input_path + "*"
+    } else {
+        smi_input_path = params.smi_input_path
+    }
+
+    if (params.fasta_path[-1] == "/") {
+        fasta_path = params.fasta_path + "*"
+    } else {
+        fasta_path = params.fasta_path
+    }
+
+    smiles = Channel.fromPath(smi_input_path)
+        .collectFile(name: 'smi_input.json')
+
+    RunMolMIMGenerate(
+        params.molmim_script,
+        params.molmim_model,
+        smiles
+    )
+
+    sequences = Channel.fromPath(fasta_path)
+        .collectFile(name: 'temp.fasta')
+        .splitFasta(record: [id: true, seqString: true])
+        .map { record -> "${record.id}zzzz${record.seqString} " }
+        .collectFile(name: 'combined_seqs.txt')
+
+    RunAlphaFold2(
+        params.alphafold2_model,
+        params.alphafold2_script,
+        sequences
+    )
+
+    num_poses = 3
+    RunDiffdock(
+        params.diffdock_script,
+        params.diffdock_model,
+        RunMolMIMGenerate.out.sdfs,
+        RunAlphaFold2.out.pdfs,
+        num_poses
+    )
+}
+
+process RunMolMIMGenerate {
+    label 'molmim'
+    cpus 8
```

Comment:
we should confirm with NVIDIA if 4/16 or 8/32 is better

Reply:
implemented a retry to go with 4/16 first then 8/32